tests/src/unit-json_patch.cpp

```diff
@@ -60,14 +60,20 @@ TEST_CASE("JSON patch")
             json const doc2 = R"({ "q": { "bar": 2 } })"_json;
 
             // because "a" does not exist.
+#if defined(JSON_DIAGNOSTIC_POSITIONS)
```

Comment:
Due to https://github.com/nlohmann/json/blob/60c48755e3f717eace7830d7bbc0d8f1a5e0cc8a/include/nlohmann/detail/abi_macros.hpp#L29-L31, `JSON_DIAGNOSTIC_POSITIONS` will always be defined. Either to `0` or `1`. So checking for `defined(JSON_DIAGNOSTIC_POSITIONS)` will always be true.

Reply:
I understand, unfortunately my workspace got messed up and I was unable to test it thoroughly, I didn't intend to push these changes, anyway I will set up a fresh workspace, I wonder if its okay to close this PR and open a new PR with fresh branch?